### ISSUE TITLE
issue to open SPM8 standalone and SPM12 standalone with the latest singularity image 

### ISSUE THREAD

**Issue by manuegrx** [brainvisa/casa-distro#268]
Hi, 

I use the last singularity image (casa-dev-5.0-4.sif ) and I cannot open SPM8Standalone or SPM12Standalone.

For SPM12 standalone I had the following error: 
```
------------------------------------------------------------------------
       Segmentation violation detected at Thu Jul  8 15:24:38 2021
------------------------------------------------------------------------
Configuration:
  MATLAB Version:   7.10.0.499 (R2010a)
  MATLAB License:   [unknown]([url](url))
  Operating System: Linux 4.4.0-186-generic #216-Ubuntu SMP Wed Jul 1 05:34:05 UTC 2020 x86_64
  GNU C Library:    2.27 stable
  Window System:    The X.Org Foundation (11804000), display :0
  Current Visual:   0x21 (class 4, depth 24)
  Processor ID:     x86 Family 6 Model 13 Stepping 7, GenuineIntel
  Virtual Machine:  Java 1.6.0_12-b04 with Sun Microsystems Inc. Java HotSpot(TM) 64-Bit Server VM mixed mode
  Default Encoding:  UTF-8
Fault Count: 1
Register State:
  rax = 0000000000000000   rbx = 00007fe6a0022230
  rcx = 0000000000000000   rdx = 00007fe6a0000080
  rbp = 0000000000000000   rsi = 000000003e07c725
  rdi = 0000000000000000   rsp = 00007fe6a5216da8
   r8 = 0000000000000064    r9 = 0000000000000060
  r10 = 00007fe6a0000080   r11 = 00007fe6a0000080
  r12 = 0000000000000000   r13 = 0000000000000000
  r14 = 00007fe6a52170c8   r15 = 0000000000000000
  rip = 00007fe6b437e426   flg = 0000000000010293
Stack Trace:
If this problem is reproducible, please submit a Service Request via:
    http://www.mathworks.com/support/contact_us/ts/help_request_1.html
A technical support engineer might contact you with further information.
Thank you for your help.
Warning: 1 invisible figure(s) exist at MCR termination.
```
For SPM8 standalone it is also a " Segmentation violation" error but it is quite long . 

I do not have theses issue when opening the same SPM standalone in my computer without casa. 
It seems that I can open SPM12 via matlab without issue. 

Is there any change in the latest image that can explain this issue ? 
Am I the only one to have trouble ? 

Emmanuelle

**Comment by denisri** [comment]
Have you checked that the problem does not happen using the `casa-dev-5.0-3.sif` image ? I'm not sure to understand everything: if the matlab MCR has changed, it means that you are using a different installation of SPM ?

**Comment by denisri** [comment]
I am only using SPM standalone for the normalization processes of Morphologist in brainvisa, and I still could run it without error on my laptop (using a rather old release of SPM12 standalone).

**Comment by Hboni** [comment]
I made more tests with SPM12 and SPM8 standalone with different images version. I tried to use a brainvisa coregister process with both versions. I also tried to open the interface of SPM inside a `bv bash` :


| | 5.0-4 | 5.0-3 | 5.0-1 |
| --- | --- | --- | --- |
| SPM8 Coreg  | error | error | OK |
| SPM12 Coreg | OK | OK | OK |
| SPM8 GUI | error | error | OK |
| SPM12 GUI | error | error | OK |

Both SPM8 and SPM12 standalone use an mcr dedicated, but with the exact same version (v713). I tried to use the mcr located in SPM12 folder to launch SPM8 (and the SPM8 mcr for SPM12). And I get the exact same results as in the table.
Furthermore I get a more verbose error with SPM8:

```
/ __)(  _ \(  \/  )                                           
\__ \ )___/ )    (   Statistical Parametric Mapping           
(___/(__)  (_/\/\_)  SPM8 - http://www.fil.ion.ucl.ac.uk/spm/


------------------------------------------------------------------------
       Segmentation violation detected at Thu Jul 15 14:46:17 2021
------------------------------------------------------------------------

Configuration:
  MATLAB Version:   7.10.0.499 (R2010a)
  MATLAB License:   unknown
  Operating System: Linux 4.13.0-43-generic #48~16.04.1-Ubuntu SMP Thu May 17 12:56:46 UTC 2018 x86_64
  GNU C Library:    2.27 stable
  Window System:    The X.Org Foundation (11804000), display :0
  Current Visual:   0x21 (class 4, depth 24)
  Processor ID:     x86 Family 6 Model 14 Stepping 9, GenuineIntel
  Virtual Machine:  Java 1.6.0_12-b04 with Sun Microsystems Inc. Java HotSpot(TM) 64-Bit Server VM mixed mode
  Default Encoding:  UTF-8

Fault Count: 1

Register State:
  rax = 0000000000000000   rbx = 00007f632404aa40
  rcx = 0000000000000000   rdx = 0000000000000000
  rbp = 0000000000000000   rsi = 000000003e07c725
  rdi = 0000000000000000   rsp = 00007f6329adabc8
   r8 = 0000000000000001    r9 = 0000000000000000
  r10 = 00007f63240008d0   r11 = 00007f6338d48514
  r12 = 0000000000000000   r13 = 0000000000000000
  r14 = 00007f6329adaee8   r15 = 0000000000000000
  rip = 00007f6338d274e1   flg = 0000000000010283

Stack Trace:

If this problem is reproducible, please submit a Service Request via:
    http://www.mathworks.com/support/contact_us/ts/help_request_1.html

A technical support engineer might contact you with further information.

Thank you for your help.


------------------------------------------------------------------------
       Segmentation violation detected at Thu Jul 15 14:46:17 2021
------------------------------------------------------------------------

Configuration:
  MATLAB Version:   7.10.0.499 (R2010a)
  MATLAB License:   unknown
  Operating System: Linux 4.13.0-43-generic #48~16.04.1-Ubuntu SMP Thu May 17 12:56:46 UTC 2018 x86_64
  GNU C Library:    2.27 stable
  Window System:    The X.Org Foundation (11804000), display :0
  Current Visual:   0x21 (class 4, depth 24)
  Processor ID:     x86 Family 6 Model 14 Stepping 9, GenuineIntel
  Virtual Machine:  Java 1.6.0_12-b04 with Sun Microsystems Inc. Java HotSpot(TM) 64-Bit Server VM mixed mode
  Default Encoding:  UTF-8

Fault Count: 2

Register State:
  rax = 00007f63399c2700   rbx = 0000000000000000
  rcx = 0000000000000001   rdx = 00007f6329add430
  rbp = 00007f6329add110   rsi = 00007f6327724a70
  rdi = 0000000000000001   rsp = 00007f6329add0d0
   r8 = 00007f6325d30d30    r9 = 0000000000000001
  r10 = 0000000000000000   r11 = 0000000000000001
  r12 = 00007f6329add2b0   r13 = 0000000000000001
  r14 = 0000000000000002   r15 = 0000000000000001
  rip = 00007f6333051954   flg = 0000000000010206

Stack Trace:
  [0] libmwm_interpreter.so:inAssignInputArgsToFormals(int, mxArray_tag**, Mfh_mp*, int)(133, 0x7f6329add370, 0, 0x7f6327724a70) + 84 bytes
  [1] libmwm_interpreter.so:inRunMfile(int, mxArray_tag**, int, mxArray_tag**, Mfh_mp*, inWorkSpace_tag*)(0x7f6327724a70, 0x13869f298, 0x7f6327548a50, 0x7f63399c2700) + 1404 bytes
  [2] libmwm_dispatcher.so:Mfh_file::dispatch_fh(int, mxArray_tag**, int, mxArray_tag**)(0x7f6329add500, 0x7f6329add6f0, 0x7f6329add870, 0x7f6327548e10) + 330 bytes
  [3] libmwmcos.so:mcos::MScopedMethodInfo::invoke(mcos::COSClient const*, int&, mcos::COSValue*, int, mcos::COSValue const*) const(0x7f63399c2170, 0x7f6329add6e0, 0x7f6329add6b0, 0x7f6329add6c0) + 349 bytes
  [4] libmwmcos.so:mcos::MDestructorMI::invoke(mcos::COSClient const*, int&, mcos::COSValue*, int, mcos::COSValue const*) const(0x7f6329add890, 0x7f6329add8b0, 0x7f6329add920, 0x7f632755e690) + 506 bytes
  [5] libmwmcos.so:mcos::COSClassI::callDestructor(mcos::COSInterfacePtr)(0x7f6329addaf0, 0x7f6327547750, 0x7f632754f080, 0x7f6329adda00) + 483 bytes
  [6] libmwmcos.so:mcos::COSClassI::destroyInstance(mcos::COSInterfacePtr)(0x7f6329add990, 0x7f632f7b9e44, 0x7f6329adda40, 0x7f632f7a912c) + 110 bytes
  [7] libmwmcos.so:mcos::COSInterfaceI::destroyUnconditional()(0x7f6329addb90, 0x7f6329addbc8, 0x7f632754f220, 0x7f62c3098280) + 1826 bytes
  [8] libmwmcos.so:mcos::COSInterfaceI::destroyRef()(0x7f62c3096a20, 0, 0x7f6329addbd0, 0x7f632f83e075) + 134 bytes
  [9] libmwmcos.so:mcos::COSInterfacePtr::release()(0x7f62c3098280, 0x7f632754f080, 0x7f6329addc10, 0x7f63381b4317) + 37 bytes
  [10] libmwmcos.so:omDestroyElementListener(void*, mxArray_tag const*)(2, 0x7f62c3098280, 17, 0x7f6327220b60) + 69 bytes
  [11] libmx.so:0x00007f63381b4317(0x7f62c3098280, 0x7f63274ab020, 0x7f6329addc60, 0x7f63381b4c4a)
  [12] libmx.so:0x00007f63381b38c0(0x7f6329addc60, 0x7f6327220bb8, 0x7f63274ab020, 0x7f6327220b60)
  [13] libmx.so:mxDestroyArray~(0x7f6329addca0, 0, 0, 0x7f63274ab020) + 74 bytes
  [14] libmwm_interpreter.so:inResetWS(inWorkSpace_tag*, int)(0xffffffff0000003f, 0x7f6329addd00, 0x274b1900, 0x7f63274ab020) + 242 bytes
  [15] libmwm_interpreter.so:inMfileEpilogFinalizer(opcodes, void*, opcodes*, int*)(0x7f6329addd90, 0x7f632778b690, 0x7f6329addf30, 0x7f632745c2d0) + 766 bytes
  [16] libmwm_interpreter.so:inRunFinalizers(opcodes, opcodes*, int*)(0x7f6329adddd0, 0x7f6329addff0, 0, 0x7f6329ade240) + 55 bytes
  [17] libmwm_interpreter.so:inRunMfile(int, mxArray_tag**, int, mxArray_tag**, Mfh_mp*, inWorkSpace_tag*)(0x7f6329ade240, 0x200000563, 0x7f632745c2d0, 0x7f6329adefc0) + 2720 bytes
  [18] libmwm_dispatcher.so:Mfh_file::dispatch_fh(int, mxArray_tag**, int, mxArray_tag**)(0x200000002, 0x7f63273f31c8 "hgload", 0x7f6329ad0000, 0x7f6327404580) + 330 bytes
  [19] libmwm_interpreter.so:inDispatchFromStack(int, char const*, int, int)(0x7f6329ade5ec, 2, 0x7f63273f31c8 "hgload", 0) + 1066 bytes

```

**Comment by denisri** [comment]
So,
- SPM12 standalone coreg (and probably normalise) do work with all images, that's probably why we didn't run into this problem earlier.
- the problem did not appear in `casa-dev-5.0-4` but in `5.0-3` or `5.0-2`.

Now after testing a bit, I will add even more confusion: I have 2 machines (a desktop and a laptop), both running Ubuntu 18.04 (with updates done), with SPM12-standalone-7771 installed (same release on both machines). Casa-distro updated to image `5.0-4`.
- on my desktop machine, SPM12 GUI starts without error in casa-distro.
- Same when run natively on the host system.
- on my laptop, I get the same crash as you get in casa-distro.
- But... **I also get the crash outside of casa-distro container** on the host system...

So would it be linked to a user config somewhere (path...) ?

**Comment by DimitriPapadopoulos** [comment]
MCR v713 is the version shipping with Matlab R2010a.

**Comment by DimitriPapadopoulos** [comment]
Perhaps https://github.com/ReproNim/neurodocker/issues/358 will help.

**Comment by DimitriPapadopoulos** [comment]
But then, do I need to update the MCR used by the standalone SPM outside of Singularity? Why hasn't this ever been reported outside Singularity here at NS?

**Comment by gllmflndn** [comment]
Hello :grinning:

I couldn't follow all your tests and I don't know what casa-distro is but there is a good chance all of the issues will disappear when using a more recent version of the MCR. R2010a was used consistently to prevent having to install a new MCR from time to time (or having to keep several versions in parallel) but it is just very dated now. I use R2019b in this [Singularity definition file](https://github.com/spm/spm-docker/blob/main/matlab/singularity.def) with no known issue.
Standalone SPM12 with R2019b is not an official release but I could try to make that happen (at least for Linux) soon if that was helpful here.

**Comment by DimitriPapadopoulos** [comment]
Hi Guillaume! So to be clear, you recommend we deploy the SPM8 and SPM12 standalone versions with the MCR v97 from Matlab R2019b? Although this ticket is about a Singularity deployment, I am personally interested in the deployment of SPM directly on Ubuntu 18.04 and 20.04 workstations, without Singularity.

If you envision a new SPM12 standalone release, why not switch directly to R2021a?

**Comment by gllmflndn** [comment]
Yes, the idea would be to release a new SPM12 standalone version (r7771) relying on the most recently available version of the MATLAB Runtime (probably waiting a few weeks for R2021b, even if ideally I was waiting for R2022a). From then on, would you prefer if new releases of SPM standalone were using the same version of the MATLAB Runtime or if they were using the most recently available one at each given time (or both)?
Also make sure to set an `SPM_HTML_BROWSER` environment variable to `0` before starting SPM as it is known otherwise to be the cause of a crash.
This plan does not concern SPM8 though - is this still of any use to you?

**Comment by DimitriPapadopoulos** [comment]
If a new MCR is required to fix bugs or crashes, by all means please do not hesitate to upgrade the MCR - that's more important than saving a few GB of disk space. On the other hand, I suggest using the same MCR as long as it works and has no security issues - every sub-system upgrade comes with its own (minim) risks of problems.

Good to learn about this [`SPM_HTML_BROWSER`](https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=SPM;e2b58a74.2001) trick for r7771.

I tried to erase SPM5, but a couple users still need it in some scripts! As you can imagine, we have even more users still in need of SPM8 here and there. One last update of SPM8 based on the latest version of Matlab would be useful. It should buy us enough time (years) to phase out SPM8 completely...

**Comment by DimitriPapadopoulos** [comment]
Has this issue been fixed? How?

**Comment by denisri** [comment]
Not that I know, but I haven't really understood if we could do anything, and what. The MCR is not in our (casa-distro) singularity image. We're running programs (brainvisa, anatomist, aims...) from a container, and SPM may get called from the container, but it's not actually installed in the container: it's up to the user to install it and set its path in brainvisa; or use the version installed on the network in noeuropspin. So unless a precise library in the casa-distro brainvisa container is incompatible with the MCR in use in neurospin, and could be changed with another one, it's not really related to casa-distro and containers ?

**Comment by DimitriPapadopoulos** [comment]
In my case, I am in the process of changing the MCR which can typically be found under `/i2bm/local/spm8-standalone-5236/mcr`.

**Comment by denisri** [comment]
So we'll see if it fixes the problems for users...

**Comment by gllmflndn** [comment]
Can you test what happens with this version of SPM12 that uses the R2021b Runtime: https://github.com/WCHN/CTseg/issues/9#issuecomment-946111042

**Comment by DimitriPapadopoulos** [comment]
Good news!

The new version:
* fixes issue 1,
* fixes issue 3, of course, since the MCR v911 from MATLAB R2021b is fully compatible with recent versions of Ubuntu such as 18.04 and 20.04,
* makes issue 4 irrelevant, as the SPM12 executable has been compiled with a recent Matlab compiler.

I'll investigate issue 2 and come back to you.

**Comment by DimitriPapadopoulos** [comment]
@Hboni and @manuegrx, could test the new Standalone SPM12 installed for testing purposes under `/i2bm/local/spm12-standalone-8168.TMP`?

Just run:
```
/i2bm/local/spm12-standalone-8168.TMP/run_spm12.sh /i2bm/local/matlab/MATLAB_Runtime/v911
```

**Comment by gllmflndn** [comment]
For issue 2, make sure the environment variable `SPM_HTML_BROWSER` is set otherwise it could explain a crash towards the end of the SPM initialisation when opening the full interface.

**Comment by Hboni** [comment]
I made some tests on my Ubuntu-18.04 computer, using several casa-distro environments. As described before, I couldn't launch SPM12 GUI on casa-distro > 5.0.1 for spm12-standalone-6472 with MCR v713.
Using spm12-standalone-8168.TMP with MCR v911, the GUI launchs correctly (with/without the `export SPM_HTML_BROWSER`).

So it seems that this new version fixes our problem! :+1:

**Comment by DimitriPapadopoulos** [comment]
It would be interesting to know what changed between older casa-distro containers and the new casa-distro > 5.0.1 containers. I suspect older containers are based on Ubuntu 16.04, while containers > 5.0.1 are based on Ubuntu 18.04.

That wouldn't really help find the detailed root cause (perhaps some incompatibility in a shared library) but at least we would know these issues are related to a change between Ubuntu 16.04 and 18.04.

**Comment by DimitriPapadopoulos** [comment]
Also you must have added a `llibXp.so.6`library to casa-distro containers, as this library is no more available any more on Ubuntu 18.04. Where did you get that library from?

**Comment by Hboni** [comment]
All the images of casa-distro that I used for tests were based on Ubuntu-18.04 [as mentionned here](https://brainvisa.info/web/download.html#installing-a-singularity-developer-environment). I could try to dig to find differences between image versions when I have some time.

**Comment by DimitriPapadopoulos** [comment]
* I am confident `libXp.so.6` will not be required by recent MCRs. I think `libXp.so.6` can be installed using the DEB package from an older Ubuntu distribution 
* We are still waiting for an official new release of SPM Standalone - 8168 is a temporary release. @gllmflndn What about a new SPM Standalone release? :smile:
* The difference in MCR sizes might be related to the above: for 8168 I have installed the full MCR download from Mathworks (`/i2bm/local/matlab/MATLAB_Runtime/v911`), while for 7771 I use the MCR bundled with SPM12 Standalone (`/i2bm/local/spm12-standalone-7771/mcr/v713`).

That said, I do see a growth in MCR sizes, but `v911` is an outlier! Here is the output of `su -h`:
```
897M	MATLAB_Compiler_Runtime/v717
1,1G	MATLAB_Compiler_Runtime/v80
1,1G	MATLAB_Compiler_Runtime/v81
1,5G	MATLAB_Compiler_Runtime/v83
1,6G	MATLAB_Compiler_Runtime/v84
1,6G	MATLAB_Compiler_Runtime/v85
1,6G	MATLAB_Compiler_Runtime/v90
8,5G	MATLAB_Runtime/v911
3,3G	MATLAB_Runtime/v93
4,1G	MATLAB_Runtime/v95
```

**Comment by denisri** [comment]
Yes, that was part of the question: is the mcr 713 bundled with SPM12 standalone a lightened version, and is it possible to lighten 911 the same way, or do we need to carry the whole 8.5 Go ? From your stats, it seems that MCRs have consistently grown in a somewhat uncontrolled exponential way (this size seems crazy to me: just the matlab runtime is 3 times the size of a complete full featured operating system with several programming languages in it - gosh !)

**Comment by denisri** [comment]
(actually I don't even know how the numbers are ordered: is 911 > 97 ?)

**Comment by DimitriPapadopoulos** [comment]
My wrong, I had misinterpreted the [MATLAB Runtime version numbers](https://fr.mathworks.com/products/compiler/matlab-runtime.html):
(911 = 9.11) > (97 = 9.7)

I have reordered the above list.

**Comment by denisri** [comment]
So we're doomed. No escape. :|

**Comment by denisri** [comment]
Unless an official version of SPM is released soon with only the needed MCR parts in it, since it is said to be possible (https://fr.mathworks.com/matlabcentral/answers/101332-why-is-matlab-runtime-so-large)..........? ;)

**Comment by gllmflndn** [comment]
I have a look at this soon but would it help already if you can specify which toolboxes to install from the MCR as we do here:
https://github.com/spm/spm-docker/blob/main/matlab/singularity.def#L22-L23

**Comment by DimitriPapadopoulos** [comment]
I don't think we can without your MATLAB Compiler. See https://fr.mathworks.com/matlabcentral/answers/101332-why-is-matlab-runtime-so-large:

> Starting from R2016a release, to distribute your application/Compiler SDK library with a subset of MATLAB Runtime that is needed for the application/library to run, you can generate an application/library installer with deploytool GUI or "compiler.package.installer" command. With this installer, the selective MATLAB Runtime will be installed together with the target standalone application or Compiler SDK library. When you generate the installer, MATLAB Runtime can be packaged either in the installer or downloaded from web.
> [...]
> MATLAB Runtime can also be installed with installer downloaded from 
> [...]
> This installer will install the monolithic runtime and there is no way to configure selective installation.

**Comment by denisri** [comment]
Well I'm not sure to be able to fully answer that ;) We are using the core SPM (normalization, segmentation...), and Dartel, but I'm not completely aware of what are using the different toolboxes of brainvisa... I don't think we want to install a "custom SPM" just for us if the main distribution ships the whole MCR in order to allow all possible toolboxes to run, if this is needed. I was just hoping the whole SPM + toolboxes could require a limited set of libraries and ship only those, but I was maybe naive.
This is probably too difficult to handle, both for you @gllmflndn for SPM, and for us in brainvisa.

**Comment by gllmflndn** [comment]
The MCR `v713` we distribute with SPM standalone is the unmodified one from 2010a. As you noticed, the MCR has evolved over time to include more and more of their (unused) toolboxes. What I was pointing out was that, despite the statement above ("This installer will install the monolithic runtime and there is no way to configure selective installation."), it seems that you can select when installing the MCR which toolboxes to deploy. See for example the final size of the singularity image:
https://github.com/spm/spm-docker/pkgs/container/spm-docker/23763165?tag=matlab

**Comment by denisri** [comment]
So you've already done that, using MCR 97 lightened to the needed toolboxes, in a Ubuntu 22.04 system. I just check it and see that the MCR there is "only" 2.3 GB (compared to 5.8 as listed by @DimitriPapadopoulos above). I did not know such a container already existed, and that's a very good news, and maybe we should rely on it. The problem is that we cannot merge singularity containers, nor call one from another one (at least I have not succeeded in such an attempt), thus any modularity seems impossible. But we could perhaps unpack the image and copy its content (all is in /opt I guess), or merge/include your singularity recipe (or part of it) into ours.
Thanks @gllmflndn ...

**Comment by gllmflndn** [comment]
I hadn't checked the actual size of the MCR there and I presume that the size mentioned in the manifest (~1.2GB) is for everything compressed. It's still quite large though. It seems that some extra files can be safely removed but I didn't look into this recently:
https://github.com/spm/spm-docker/pull/2
From what I understand you are doing, I would just copy the relevant bit of the singularity recipe to install only some parts of the MCR:
https://github.com/spm/spm-docker/blob/main/matlab/singularity.def#L18-L25

**Comment by denisri** [comment]
Yes, singularity .sif images are a bit compressed, and are larger inside than outside (the magic "bag of holding" concept). Yes it seems that for SPM you install just the MCR core and "numeric" package. It is still big, yes, but less than the full thing. And, yes, it's probably the good way to go, thanks again :)

**Comment by denisri** [comment]
By the way, @DimitriPapadopoulos, we should probably install this version of SPM12 (7771 + v97) in neurospin, shouldn't we ?

**Comment by DimitriPapadopoulos** [comment]
Mix 7771 with v97? But that cannot work, can it?

We have:
* 7771 with the bundled v713
* temporary test version 8168 with v97

**Comment by gllmflndn** [comment]
Would [this ](https://www.fil.ion.ucl.ac.uk/spm/download/restricted/utopia/dev/spm12_r7771_Linux_R2019b.zip)help?

**Comment by denisri** [comment]
It seems to: that's what is shipped in the SPM containers (it's obviously another build of 7771 based on v97, not the one we are using in neurospin). I have done it on my laptop and it works, just as in https://github.com/spm/spm-docker/blob/main/matlab/singularity.def

**Comment by denisri** [comment]
(otherwise I could not run the SPM 8168 installed in neurospin with a MCR 911 installed "minimally": it complains that the MCR version is not the expected one, whereas it does work with the "full" MCR 911 installed in neurospin (same version, in principle?), and I don't understand that - but maybe I did a mistake somewhere)

**Comment by denisri** [comment]
> temporary test version 8168 with v97

I just realize I did not read that correctly - 8186 is built with v97, not 911 ? I ran it using 911 and it did not complain (the full 911, as I said just earlier)

**Comment by DimitriPapadopoulos** [comment]
My wrong. We currently have:
* 7771 with the bundled MCR v713
* temporary test version 8168 with v911

I will install:
* 7771 with MCR v97, as suggested by @gllmflndn (I assume I just need to overwrite with the ZIP file and point to our v97 instead of v713

**Comment by DimitriPapadopoulos** [comment]
> I just realize I did not read that correctly - 8186 is built with v97, not 911 ? I ran it using 911 and it did not complain (the full 911, as I said just earlier)

8186 is built with v911. From `/i2bm/local/_spm12-standalone-8168/spm12_mcr/.META/manifest.xml`:
```xml
<manifest ... mcr-major-version="9" mcr-minor-version="11" mcr-release-version="0" need_exclusive_mcr_process="0" need_java="1 ...>
```
You might find a list of required MCR components in there.

**Comment by denisri** [comment]
I think the installation of spm12-standalone-7771 + mcr v97 fixes this issue. I guess we can close the ticket now.
Thanks to all.

**Comment by mfschmidt** [comment]
I struggled a lot with trying to get spm12 and a matlab runtime into a modern docker container. But I stumbled across the [SPM spm-docker github project](https://github.com/spm/spm-docker) with a very useful and automatically-updating method in the [matlab/Dockerfile](https://github.com/spm/spm-docker/blob/main/matlab/Dockerfile). Huge thanks to SPM's developers for doing this. My local adaptation of that recipe works every time and saves me from spending hours figuring out what to break in a modern OS to get a decades-old matlab runtime working.
